src/traverser.js

```diff
@@ -43,10 +43,17 @@ class Traverser {
 
 const handleFile = async (file, filePath, src, dest, dry) => {
   const splitFilePath = filePath.split('.')
-  const filePathNoExt = splitFilePath[0]
-  const srcExt = splitFilePath[1]
 
-  if (srcExt === src) {
+  let match = true
+  for (let i = src.length - 1; i >= 0 && match; i--) {
```

Comment:
Can we make this logic a bit easier to grok? 
- Maybe a `while` loop as opposed to a `for` loop counting down and also based on truthiness of `match`? 
- Also the fact that we are mutating the `splitFilePath` array and then using it I think can easily be missed. Perhaps we do not do mutations and build up a new array. What do you think?

Reply:
Continuing from above:
Let `src` and `dist` be the plain inputted strings and `fileName` the files name.
```js
const extension = fileName.slice(-src.length)
if (extension === src) { // match
  const filePathNoExt = path.slice(0, -src.length)
  // renaming but without the added dots, e.g.
  const oldDestinationPath = `${filePathNoExt}.{dest}`
  const newDestinationPath = filePathNoExt + dest
}
```
This would also allow for renaming of entire files (e.g. `rename myfile.txt yourfile.doc`), with the caveat of also matching a file named `txt` when setting src to `txt` instead of `.txt` (would require a major version bump).
If we agree on taking this approach we should also display a warning in case `src` __xor__ `dest` start with a dot.

Reply:
I like this. Seems a lot simpler and less error prone, but for now lets just stick to the extension piece

Reply:
Are you sure?
If you want to keep the current logic and make it simpler I'd go for something along those lines:

```js
const pathFragments = path.split('.')
const pathNoExt = pathFragments.slice(0, -srcFragments.length).join('.')
/* renamed src to srcFragments in order to clarify it being split */
const extFragments = pathArray.slice(-srcFragments.length)

let match = true
for (let i = 0; i < srcFragments.length && match; i++) {
  match = extFragments[i] === srcFragments[i]
}
```

However I strongly favour the approach from my last post, let me know what you think. Its behaviour would also be better in line with the package name 👍maybe in conjunction with a wildcard operator, like that

| File name | src | match |
| ---------- | --- | ------ |
| test.js        | s     | __-__   |
| test.js        | js    | ✓        |
| test.js        | .js   | ✓        |
| test.js        | t.js  | __-__   |
| test.js        | *t.js | ✓        |

Reply:
Ok I am convinced. Let's go with the first way. Also if it makes #4 easy that is a nice win